tests/Console/Command/DescribeCommandTest.php

```diff
@@ -0,0 +1,155 @@
+{
+    public function testExecute()
+    {
+        $fixer = $this->getMockBuilder('PhpCsFixer\Tests\Console\Command\DefinedAndConfigurableFixerInterface')->getMock();
+        $fixer
+            ->method('getName')
+            ->willReturn('Foo/bar')
+        ;
+        $fixer
+            ->method('getDefinition')
+            ->willReturn(new FixerDefinition(
+                'Fixes stuff.',
+                array(
+                    new CodeSample(
+                        '<?php echo \'bad stuff and bad thing\';'
+                    ),
+                    new CodeSample(
+                        '<?php echo \'bad stuff and bad thing\';',
+                        array('things' => true)
+                    ),
+                ),
+                'Replaces bad stuff with good stuff.',
+                'Option "things" enables fixing things as well.',
+                array('things' => false),
+                'Can break stuff.'
+            ))
+        ;
+        $things = false;
+        $fixer
+            ->method('configure')
+            ->willReturnCallback(function (array $configuration = null) use (&$things) {
+                $things = !empty($configuration['things']);
+            })
+        ;
+        $fixer
+            ->method('fix')
+            ->willReturnCallback(function (\SplFileInfo $file, Tokens $tokens) use (&$things) {
+                $tokens[3]->setContent($things ? '\'good stuff and good thing\'' : '\'good stuff and bad thing\'');
+            })
+        ;
+
+        $fixerFactory = new FixerFactory();
+        $fixerFactory->registerFixer($fixer, true);
+
+        $application = new Application();
+        $application->add(new DescribeCommand($fixerFactory));
+
+        $command = $application->find('describe');
+
+        $commandTester = new CommandTester($command);
+        $commandTester->execute(array(
+            'command' => $command->getName(),
+            'name' => 'Foo/bar',
+        ));
+
+        $expected = <<<'EOT'
+Description of Foo/bar rule.
+Fixes stuff.
+Replaces bad stuff with good stuff.
+
+Fixer is configurable.
+Option "things" enables fixing things as well.
+Default configuration: ['things' => false].
+
+Fixing examples:
+ * Example #1.
+   ---------- begin diff ----------
+   --- Original
+   +++ New
+   @@ @@
+   -<?php echo 'bad stuff and bad thing';
+   +<?php echo 'good stuff and bad thing';
+%spaces%
+   ----------- end diff -----------
+
+ * Example #2. Fixing with configuration: ['things' => true].
+   ---------- begin diff ----------
+   --- Original
+   +++ New
+   @@ @@
+   -<?php echo 'bad stuff and bad thing';
+   +<?php echo 'good stuff and good thing';
+%spaces%
+   ----------- end diff -----------
+
+
+EOT;
+
+        $this->assertSame(
+            str_replace(array('%spaces%', "\n"), array('   ', PHP_EOL), $expected),
+            $commandTester->getDisplay()
+        );
+
+        if (method_exists($commandTester, 'getStatusCode')) {
```

Comment:
Would be better to make this test it's own test method, so we can actually mark it as skipped properly, instead of silently not testing it.